engine/apps/alerts/tasks/notify_ical_schedule_shift.py

```diff
@@ -183,160 +26,137 @@ def notify_ical_schedule_shift(schedule_pk):
+    shift_changed, diff_shifts = calculate_shift_diff(current_shifts, prev_shifts)
 
-        upcoming_shifts = []
-        # Add the earliest next_shift
-        if len(next_shifts_from_ical) > 0:
-            earliest_shift = next_shifts_from_ical[0]
-            upcoming_shifts.append(earliest_shift)
-            # Check if there are next shifts with the same start as the earliest
-            for shift in next_shifts_from_ical[1:]:
-                if shift["start"] == earliest_shift["start"]:
-                    upcoming_shifts.append(shift)
-
-        empty_oncall = len(current_shifts) == 0
-        if empty_oncall:
-            schedule.empty_oncall = True
-        else:
-            schedule.empty_oncall = False
+    # Do not notify if there is no difference between current and previous shifts
+    if not shift_changed:
+        task_logger.info(f"No shift diff found for schedule {schedule_pk}, organization {schedule.organization_id}")
+        # If prev shifts were converted to a new format, update related field in db
+        if prev_shifts_updated:
             schedule.current_shifts = json.dumps(current_shifts, default=str)
+            schedule.save(update_fields=["current_shifts"])
+        return
 
-        schedule.save(update_fields=["current_shifts", "empty_oncall"])
-
-        if len(new_shifts) > 0 or empty_oncall:
-            task_logger.info(f"new_shifts: {new_shifts}")
+    # Get only new/changed shifts to send a reminder message.
+    new_shifts = sorted(diff_shifts, key=lambda shift: shift["start"])
+
+    # get days_to_lookup for next shifts
+    if len(new_shifts) != 0:
+        days_to_lookup = (new_shifts[-1]["end"].date() - now.date()).days + 1
+        days_to_lookup = max([days_to_lookup, MIN_DAYS_TO_LOOKUP_FOR_THE_END_OF_EVENT])
+    else:
+        days_to_lookup = MIN_DAYS_TO_LOOKUP_FOR_THE_END_OF_EVENT
+
+    datetime_end = now + datetime.timedelta(days=days_to_lookup)
+
+    next_shifts_unfiltered = schedule.final_events(now, datetime_end, False, False)
+    # drop events that already started
+    next_shifts = []
+    for next_shift in next_shifts_unfiltered:
+        if now < next_shift["start"]:
+            next_shifts.append(next_shift)
+
+    next_shifts = sorted(next_shifts, key=lambda shift: shift["start"])
+
+    upcoming_shifts = []
+    # Add the earliest next_shift
+    if len(next_shifts) > 0:
+        earliest_shift = next_shifts[0]
+        upcoming_shifts.append(earliest_shift)
+        # Check if there are next shifts with the same start as the earliest
+        for shift in next_shifts[1:]:
+            if shift["start"] == earliest_shift["start"]:
+                upcoming_shifts.append(shift)
+
+    empty_oncall = len(current_shifts) == 0
+    if empty_oncall:
+        schedule.empty_oncall = True
+    else:
+        schedule.empty_oncall = False
+        schedule.current_shifts = json.dumps(current_shifts, default=str)
```

Comment:
nit, I think this could be changed to something like
```
schedule.empty_oncall = len(current_shifts) == 0
if not schedule.empty_oncall:
    schedule.current_shifts = json.dumps(current_shifts, default=str)
```
?